aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java

```diff
@@ -73,6 +81,62 @@
                         requestLogger));
     }
 
+    protected ProgressEvent<ResourceModel, CallbackContext> updateTags(
+            final ProxyClient<RdsClient> rdsProxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final Tagging.TagSet previousTags,
+            final Tagging.TagSet desiredTags
+    ) {
+        final Tagging.TagSet tagsToAdd = Tagging.exclude(desiredTags, previousTags);
+        final Tagging.TagSet tagsToRemove = Tagging.exclude(previousTags, desiredTags);
+
+        if (tagsToAdd.isEmpty() && tagsToRemove.isEmpty()) {
+            return progress;
+        }
+
+        final String arn = progress.getCallbackContext().getDbParameterGroupArn();
+
+        try {
+            Tagging.removeTags(rdsProxyClient, arn, Tagging.translateTagsToSdk(tagsToRemove));
+            Tagging.addTags(rdsProxyClient, arn, Tagging.translateTagsToSdk(tagsToAdd));
+        } catch (Exception exception) {
+            return Commons.handleException(
+                    progress,
+                    exception,
+                    SOFT_FAIL_NPROGRESS_TAGGING_ERROR_RULE_SET
+            );
+        }
+
+        return progress;
+    }
+
+    protected CallChain.Completed<DescribeDbParameterGroupsRequest, DescribeDbParameterGroupsResponse, RdsClient, ResourceModel, CallbackContext>
```

Comment:
Nit: the return type looked so ugly that I decided to introduce a bit of code duplication.

Reply:
Removed it